### DISCUSSION TITLE
Use with C-style arrays

### DISCUSSION THREAD

**Discussion by saporta** [beached/daw_json_link#354] [Q&A] [answered]
Hi :)
I'm having a hard time figuring out how to use this library with C-style arrays:
struct foo{
  int bar[5]
}

I couldn't find any "regular" way with json_array, and I failed to create a working custom constructor :(.
Also no one online had asked or shared something about using C arrays, and it makes me wonder is it that much of a rare use-case or am I missing something so obvious that it doesn't even need to be in the documentation lol.

**Comment by beached** [comment]
This should work, I added an example in the tests/src/carray_test.cpp .  There was an issue that https://github.com/beached/daw_json_link/pull/356 fixes and should be released soon.  carray's are not going to be optimal because they cannot be returned from a function, if you can use a std::array it is much simpler and just works.  

The example parses to a std::array and then copies the elements to the c-array submember.  This must be done in the context of the owning classes construction via a custom constructor so that it can be returned.

**Comment by beached** [comment, marked as answer]
This should give an idea of what is needed to use c-arrays
https://github.com/beached/daw_json_link/blob/release/tests/src/carray_test.cpp

**Reply by saporta** [reply]
Thanks a lot!
